### ISSUE TITLE
Swift 4 issue in sublabelSize()

### ISSUE THREAD

**Issue by florianweich** [cbpowell/MarqueeLabel#196]
### Usage Details
  - Language: Swift
  - Platform: Apple TV
  - tvOS Version: 11 beta 7
  - Xcode Version: 9 beta 6

I converted my project to Swift 4 using the current Xcode Beta. When setting text in the MarqueeLabel the app crashes in `sublabelSize() -> CGSize` on line 721  
`var expectedLabelSize = sublabel.sizeThatFits(maximumLabelSize)`  
with a BAD_ACCESS exception:
`Thread 1: EXC_BAD_ACCESS (code=2, address=0x16fb2bdf4)`

*UPDATE*
This issue only happens when .animationDelay ist set to 0 (regardless of being set from code or Interface Builder).

**Comment by cbpowell** [comment]
Interesting, I'll check it out. I hadn't gotten around to doing the Swift 4 conversion just yet.

Did you happen to test with tvOS 11 prior to the Swift conversion? `sizeThatFits` is a UIKit function, and that's a pretty plain usage. I wonder if tvOS/iOS 11 changed something? Maybe even with the `maximumLabelSize` variable that's used as an argument?

**Comment by florianweich** [comment]
The old binary works just fine on tvOS 11. This issue occurs only when building against the beta SDK.
I also created a complete new project with just one label in the view. Same thing there when the animationDelay is set to 0.

**Comment by cbpowell** [comment]
Hmm I'm not able to replicate it with the demo project converted to Swift 4, using the latest Xcode & SDK, or with a new (Swift 4) tvOS project importing only one view. So unfortunately I'm not sure what's causing your issue.

Would you be willing to upload that new project you made that also exhibits the problem so I could try it?

**Comment by florianweich** [comment]
Sorry for my delayed answer! I uploaded the project here: https://github.com/florianweich/LabelTest
This project also crashes on the newly released Beta 8 of tvOS.

**Comment by cbpowell** [comment]
Thanks for uploading that! It helped me pin it down. (Also sorry for the delayed answer on my side)

So it looks like it was actually an issue with using a `.rate` type of label speed. When using a rate, the `animationDuration` property needs to be calculated based on the distance the label needs to scroll at that rate, but with the changes I made to implement the `ScrollSequence` stuff I had the calculation order backwards. The `animationDuration` would be calculated based on the default offset value of `0.0` because the true offset wasn't set until later in the code (probably should have leveraged Swift and made that an Optional value!). I made `animationDuration` a calculated property to correct the issue, so any time its used it should be the up-to-date value.
